x-pack/filebeat/input/awss3/input.go

```diff
@@ -376,5 +380,16 @@ func getProviderFromDomain(endpoint string, ProviderOverride string) string {
 	return "unknown"
 }
 
+func PollSqsWaitingMetric(ctx context.Context, receiver *sqsReader) {
+	t := time.NewTicker(1 * time.Minute)
+
+	for range t.C {
```

Comment:
There should be some way here for the context to propagate its cancellation to the ticker. Maybe something like
```suggestion
	t := time.NewTicker(time.Minute)

	for range t.C {
		select {
		case <-ctx.Done():
			t.Stop()
			return
		default:
		}
```